src/main/java/com/autotune/analyzer/recommendations/model/CostBasedRecommendationModel.java

```diff
@@ -577,11 +582,32 @@ public Map<AnalyzerConstants.RecommendationItem, RecommendationConfigItem> getAc
             return null;
         }
 
-        double coreAverage = CommonUtils.percentile(COST_ACCELERATOR_PERCENTILE, acceleratorCoreMaxValues);
-        double memoryAverage = CommonUtils.percentile(COST_ACCELERATOR_PERCENTILE, acceleratorMemoryMaxValues);
+        // Return null if entries are empty
+        if (acceleratorCoreMaxValues.isEmpty() && acceleratorMemoryMaxValues.isEmpty())
+            return null;
+
+        double coreAverage = 0.0;
+        if (!acceleratorCoreMaxValues.isEmpty())
+            coreAverage = CommonUtils.percentile(COST_ACCELERATOR_PERCENTILE, acceleratorCoreMaxValues);
+
+        double memoryAverage = 0.0;
+        if (!acceleratorMemoryMaxValues.isEmpty())
+            memoryAverage = CommonUtils.percentile(COST_ACCELERATOR_PERCENTILE, acceleratorMemoryMaxValues);
```

Comment:
Looks like redundant checks as it has already been checked in line 586. Please remove them

Reply:
Thanks for the review @dinogun . In the earlier check we `continue` if both of the lists are empty. at the time of calculating percentile we check if the respective list is not empty. So I feel we will be needing this check.